supervisor/archives_api.go

```diff
@@ -92,17 +92,25 @@ func (self ArchiveAPI) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
+		taskChan := make(chan string)
 		// tell the supervisor to schedule a task
 		self.Tasks <- &db.Task{
-			Op:          db.RestoreOperation,
-			Owner:       params.Owner,
-			TargetUUID:  tid,
-			ArchiveUUID: id,
-			RestoreKey:  archive.StoreKey,
+			Op:           db.RestoreOperation,
+			Owner:        params.Owner,
+			TargetUUID:   tid,
+			ArchiveUUID:  id,
+			RestoreKey:   archive.StoreKey,
+			TaskUUIDChan: taskChan,
 		}
 
-		w.WriteHeader(200)
-		JSONLiteral(w, fmt.Sprintf(`{"ok":"scheduled"}`))
+		var taskUUID string = <-taskChan
+		if taskUUID == "" {
+			w.WriteHeader(500)
+			JSONLiteral(w, fmt.Sprintf(`{"ok":"error"}`))
+		} else {
+			w.WriteHeader(200)
+			JSONLiteral(w, fmt.Sprintf(`{"ok":"scheduled","task_uuid":"%s"}`, taskUUID))
```

Comment:
Do we use `JSONLiteral` like this elsewhere?  If not, we should use real JSON marshaling.  If so, we should put a ticket in to use real JSON marshaling globally.